hollow/src/main/java/com/netflix/hollow/core/read/engine/object/HollowObjectTypeReadStateShard.java

```diff
@@ -314,24 +315,18 @@ private long fieldOffset(HollowObjectTypeDataElements currentData, int ordinal,
      * @param out
      * @return
      */
-    private static final ThreadLocal<char[]> chararr = new ThreadLocal<char[]>();
+    private static final ThreadLocal<char[]> chararr = ThreadLocal.withInitial(() -> new char[100]);
 
     private String readString(ByteData data, long position, int length) {
-        long endPosition = position + length;
-
-        char chararr[] = getCharArray();
-
-        if(length > chararr.length)
+        char[] chararr = HollowObjectTypeReadStateShard.chararr.get();
+        if (length > chararr.length) {
             chararr = new char[length];
```

Comment:
Is there a reason to not set the threadlocal `chararr` to this new bigger array?

Reply:
Only to retain existing behaviour, I'm not sure what the original rationale was.

Reply:
May have been an oversight, what do you think @dkoszewnik ?

Reply:
Not an oversight -- the rationale is that we don't want an occasional extremely long string to cause the ThreadLocal to hang on to an extremely large buffer.